tools/apiview/parsers/apiview-treestyle-parser-schema/models.tsp

```diff
@@ -0,0 +1,150 @@
+model CodeFile {
+  /** The version of the parser */
+  VersionString: string;
+
+  /** Name the ocde file. Usually just the package name and version*/
+  Name: string;
+  
+  Language: string;
+  LanguageVariant: "Default" | "Spring" | "Android";
+  PackageName: string;
+  ServiceName: string;
+  PackageVersion: string;
+
+  /** Maps related packages across Languages */
+  CrossLanguagePackageId: string;
+
+  /** The API Tree(s) */
+  APIForest: APITreeNode[];
+
+  Diagnostics: CodeDiagnostic[];
+}
+
+model CodeDiagnostic {
+  DiagnosticId: string;
+  Text: string;
+  HelpLinkUri: string;
+
+  /** Maps to the APITreeNode id */
+  TargetId: string;
+  Level: CodeDiagnosticLevel;
+}
+
+model APITreeNode {
+  /** The name of the tree node which will be used as label for the API Navigation. Generally this is the name of the module (class, method). */
+  Name: string;
+
+  /** Id of the node, which should be unique at the node level. i.e. unique among its siblings. This was previously represented by the DefinitionId for the main Token of the node. */
+  Id: string;
+
+  /** The kind of node */
+  Kind: "assembly" | "namespace" | "class" | "delegate" | "enum" | "interface" | "method" | "package" | "struct" | "type";
+
+  /** Tokens which are rendered after all child nodes. Depending on the language this would include the closing curly brace and/or empty lines. */
+  BottomTokens?: StructuredToken[];
+
+  /** The nodes immediate children. For a namespace this would be classes, for a class this would be the class constructors and methods. Children are rendered after TopTokens but before BottomTokens, and are automatically indented.*/
+  Children?: APITreeNode[];
+
+  /** Properties of the APITreeNode. */
+  Properties?: APITreeNodeProperties;
+
+  /** Tags of the APITreeNode. */
+  Tags?: APITreeNodeTags[];
+}
+
+model StructuredToken {
+  /** Token Id. Previously known as DefinitionId. */
```

Comment:
Is this required to be unique? Is this the ID where comments are attached?

Reply:
Its not required to be unique, there are many cases where it's not. The effect is that the comment is added twice on the page.

Reply:
> The effect is that the comment is added twice on the page.

I've yet to see any situation where this was viewed as a useful feature rather than an outright logic bug. So much so that in many of my parser tests I do test that all ids are unique and fail the test if they aren't. Regardless of whether it _needs_ to be unique, if it _should_ be unique, we should document that. You also should document that it needs to be deterministic (i.e. it cannot be some random UUID).

Reply:
The same token can occur more than once on the page. I agree that making it unique will be ideal I will prefer to have more discussion about this given it changes existing behavior. @praveenkuttappan  what do you think?

Reply:
Well, this is just a question of the docstring, not the implementation.